### PR TITLE
[Feature][MRV2][Spec Decode] support synthetic rejection sampling on MRV2

### PR DESCRIPTION
**Pull request by SOMEONEUNSEEN**
## What this PR does / why we need it?
Syncs upstream vLLM [#41035](https://github.com/vllm-project/vllm/pull/41035) to the MRV2 NPU rejection sampler, enabling speculative_config.rejection_sample_method="synthetic" on Ascend.

What it does:

- _probabilistic_rejection_kernel gains a SYNTHETIC_MODE constexpr and a synthetic_conditional_rates_ptr input. In synthetic mode, both the greedy and probabilistic paths replace the standard probability-ratio acceptance test with u < rate , where the conditional rates are derived from the user-configured synthetic_acceptance_rates . Log-sum-exp is still computed for the rejected-token resampling path.
- Host function rejection_sample() accepts synthetic_conditional_rates: torch.Tensor | None and forwards it to the kernel, removing the previous hard limitation (synthetic mode was unreachable on NPU).
NPU adaptation (semantics preserved, no upstream code path touched):

- Upstream tl_rand32 is unavailable on triton-ascend; u is generated with the existing 1-element-block tl.rand + clamp(2^-31) pattern already used elsewhere in this kernel, which is equivalent to includes_zero=False semantics.
Why we need it: synthetic rejection sampling decodes without draft-model probability verification, trading exact acceptance for configurable acceptance rates — required for dynamic speculative-length methods (e.g. dspark budget control) and for throughput tuning on models where draft logits are unavailable. Without this sync, the config combination silently falls into unsupported territory on NPU.

## Does this PR introduce any user-facing change?
Yes — new capability, no behavior change for existing users:

- Users can now serve with --speculative-config '{"method": ..., "rejection_sample_method": "synthetic", "synthetic_acceptance_rates": [...]}' on Ascend MRV2. Previously this raised NotImplementedError -class failures.
- Default behavior ( rejection_sample_method="standard" ) is unchanged: synthetic_conditional_rates=None keeps SYNTHETIC_MODE=False and the kernel follows the exact pre-existing code path.
## How was this patch tested?
- Unit/e2e test added: test_synthetic_rejection_sample in tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_rejection_sample_v2.py (ported from upstream), asserting that per-position acceptance rates match the configured unconditional rates within 1e-2, across 8 parameterized cases: greedy (temp=0) and sampling (temp=1.0), rates at extremes (all-1.0 / all-0.0 / mixed), and num_speculative_steps 1 and 3. Draft tokens are multinomial-sampled on CPU (NPU multinomial reliability caveat) and moved to device.
```
tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_rejection_sample_v2.py::test_synthetic_rejection_sample[3-1.0-unconditional_rates0] [W903 10:53:21.086648810 NPUCachingAllocator.cpp:202] Warning: The current CANN and HDK(driver) versions require processing for 32 padding size, with memory allocation. (function operator())
[10:53:23]
PASSED
tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_rejection_sample_v2.py::test_synthetic_rejection_sample[3-0.0-unconditional_rates1] [10:53:24]
PASSED
tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_rejection_sample_v2.py::test_synthetic_rejection_sample[3-1.0-unconditional_rates2] [10:53:26]
PASSED
tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_rejection_sample_v2.py::test_synthetic_rejection_sample[3-0.0-unconditional_rates3] [10:53:27]
PASSED
tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_rejection_sample_v2.py::test_synthetic_rejection_sample[3-1.0-unconditional_rates4] [10:53:29]
PASSED
tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_rejection_sample_v2.py::test_synthetic_rejection_sample[3-0.0-unconditional_rates5] [10:53:30]
PASSED
tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_rejection_sample_v2.py::test_synthetic_rejection_sample[1-1.0-unconditional_rates6] [10:53:37]
PASSED
tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_rejection_sample_v2.py::test_synthetic_rejection_sample[1-0.0-unconditional_rates7] [10:53:44]
PASSED[INFO] Model cache cleared
```

`'speculative_config': {'num_speculative_tokens': 3, 'method': 'mtp', 'enforce_eager': True, 'rejection_sample_method': 'synthetic', 'synthetic_acceptance_rates': [0.9, 0.5, 0.2]}`
```
============ Serving Benchmark Result ============
Successful requests:                     64        
Failed requests:                         0         
Maximum request concurrency:             16        
Benchmark duration (s):                  156.05    
Total input tokens:                      65536     
Total generated tokens:                  131072    
Request throughput (req/s):              0.41      
Output token throughput (tok/s):         839.93    
Peak output token throughput (tok/s):    416.00    
Peak concurrent requests:                22.00     
Total token throughput (tok/s):          1259.90   
---------------Time to First Token----------------
Mean TTFT (ms):                          3525.15   
Median TTFT (ms):                        857.70    
P99 TTFT (ms):                           12653.47  
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          17.23     
Median TPOT (ms):                        17.26     
P99 TPOT (ms):                           20.67     
---------------Inter-token Latency----------------
Mean ITL (ms):                           44.78     
Median ITL (ms):                         39.51     
P99 ITL (ms):                            253.25    
---------------Speculative Decoding---------------
Acceptance rate (%):                     53.32     
Acceptance length:                       2.60      
Drafts:                                  50419     
Draft tokens:                            151257    
Accepted tokens:                         80648     
Per-position acceptance (%):
  Position 0:                            90.07     
  Position 1:                            49.97     
  Position 2:                            19.91     
```

- Lint: ruff check + ruff format + py_compile pass.
- Suggested CI verification: nightly single-card run of the above test file; no multi-card or serving-level test is required for this change.

- vLLM main: https://github.com/vllm-project/vllm/commit/ba07e4a48fc951300d97eb506217dd530583dea3

### CODE REVIEW

File: vllm_ascend/worker/v2/spec_decode/rejection_sampler_utils.py
Comment (AuroraEmiya) [inline vllm_ascend/worker/v2/spec_decode/rejection_sampler_utils.py:1]: Move new ops implementation into vllm_ascend/ops/triton/v2/spec_decode/ and add a doc

Comment (SOMEONEUNSEEN) [reply vllm_ascend/worker/v2/spec_decode/rejection_sampler_utils.py]: Thank you for your suggestion. I have added a doc. I think it is still better to keep the file path consistent with the upstream.

**Review by ZT-AIA — APPROVED** [review]

**Review by yiz-liu — APPROVED** [review]

**Review by ella1107 — APPROVED** [review]
